### PR TITLE
Kazoo 3572

### PR DESCRIPTION
**Pull request by macpie**
As a User, I'd like to be able to lower per-minute credit amount (ie. from $10 to $5), so that I can manage sub accounts per-minute credit balances.

### CODE REVIEW

**Comment by k-anderson** [comment]
Please backport to 3.20 if accepted.

File: applications/crossbar/src/modules/cb_transactions.erl [hunk 1]
```diff
@@ -101,8 +113,66 @@ validate(Context) ->
...
+%% @public
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+delete(Context, ?DEBIT) ->
+    case cb_context:resp_status(Context) of
+        'success' ->
+            case create_debit_tansaction(Context) of
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_transactions.erl]: Let's put this clause in a function, perhaps called `maybe_create_debit_transaction(Context)`

File: applications/crossbar/src/modules/cb_transactions.erl [hunk 2]
```diff
@@ -101,8 +113,66 @@ validate(Context) ->
...
+                {'ok', Transaction} ->
+                    cb_context:set_resp_data(
+                        Context
+                        ,wh_transaction:to_public_json(Transaction)
+                    )
+            end;
+        _RespStatus -> Context
+    end.
+
+create_debit_tansaction(Context) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_transactions.erl]: Spec

File: applications/crossbar/src/modules/cb_transactions.erl [hunk 3]
```diff
@@ -101,8 +113,66 @@ validate(Context) ->
...
+    Amount = wh_json:get_float_value(<<"amount">>, JObj),
+    Units = wht_util:dollars_to_units(Amount),
+    Meta =
+        wh_json:from_list([
+            {<<"auth_account_id">>, cb_context:auth_account_id(Context)}
+        ]),
+    Routines = [
+        fun(Tr) -> wh_transaction:set_reason(<<"admin_discretion">>, Tr) end
+        ,fun(Tr) -> wh_transaction:set_metadata(Meta, Tr) end
+        ,fun(Tr) -> wh_transaction:save(Tr) end
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_transactions.erl]: `,fun wh_transaction:save/1` is enough here

File: applications/crossbar/src/modules/cb_transactions.erl [hunk 4]
```diff
@@ -111,29 +181,102 @@ validate_transactions(Context, ?HTTP_GET) ->
...
+                ,wh_json:from_list([
+                    {<<"message">>, Message}
+                    ,{<<"cause">>, FuturAmount}
+                ])
+                ,Context
+            )
+    end.
+
+-spec current_account_dollars(ne_binary()) -> float().
+current_account_dollars(Account) ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_transactions.erl]: I think this function is in the teletype-related templates too. Perhaps `wht_util` should be the home for this function?

**Comment by fenollp** [comment]
rebase to fix travis

**Pull request merged**